### ISSUE TITLE
Multi-label question classifier performs poorly for combination of questions

### ISSUE THREAD

**Issue by dannypz97** [keras-team/keras#12604]
I'm trying to build a multi-label question classifier that determines all suitable labels for a question.
Currently, I'm experimenting with the [TREC-6 dataset](http://cogcomp.org/Data/QA/QC/). I've combined the 5500 question training set and the test set for training my model.
I've experimented with dozens of models, and I'm facing more or less the same issue with each of them, i.e., the model classifies questions with a single label perfectly, but when you combine 2 or more questions, it gives a good label for only the first part of the question or identifies other labels with a very low confidence.

```
E.g.: For a simple question, "where is China located?", model applies a label 'location'.
For a simple question, "who is the father of China?", model applies a label 'human'.
But for a more complex question like "where is China located and who is the father of China?",
it identifies only 'location' as a label with high confidence, whereas it 
should identify 'human' as well.
```

Currently, I'm experimenting with BiLSTMs and GLoVE embeddings, although I've tried CNNs and C-LSTMS, etc. All have more or less the same problem. There are some cases where CNNs seemed to work better but those are very few.  Moreover, my teacher discouraged me from using CNNs for some vague reason. My current architecture is somewhat like

```
inputs = Input(shape=(self.MAX_SEQUENCE_LENGTH,), dtype='int32')
embedding = Embedding(input_dim=(len(self.WORD_INDEX) + 1), output_dim=self.EMBEDDING_DIM,
                                weights=[embedding_matrix],
                                input_length=self.MAX_SEQUENCE_LENGTH, trainable=False)(inputs)

X = keras.layers.SpatialDropout1D(0.2)(embedding)      
        
X = keras.layers.Bidirectional(keras.layers.CuDNNGRU(5,return_sequences=True))(X)
X = keras.layers.Bidirectional(keras.layers.CuDNNGRU(4,return_sequences=True))(X)
        
X = Dropout(0.5)(X)
        
max_pool = keras.layers.GlobalMaxPooling1D()(X)
avg_pool = keras.layers.GlobalAveragePooling1D()(X)
attn = AttentionWeightedAverage()(X)
last = keras.layers.Flatten()(keras.layers.Lambda(lambda x: x[:,-1:,:])(X))
        
X = Concatenate()([max_pool, avg_pool, attn, last])
        
#crf = CRF(12, learn_mode='marginal')
#X = crf(X)
#X = keras.layers.Lambda(lambda x: keras.backend.expand_dims(x, axis=-1))(X)       
#X = keras.layers.Bidirectional(keras.layers.CuDNNGRU(12,return_sequences=True))(X)
X = Dropout(0.4)(X)
X = keras.layers.BatchNormalization()(X)
X = keras.layers.Dense(3, activation='relu')(X)       
        
output = keras.layers.Dense(units=self.LABEL_COUNT, activation='sigmoid')(X)
#output = keras.layers.Lambda(lambda x: x[:,1,:])(output)
model = Model(inputs=inputs, outputs=output, name='intent_classifier')
```

I've tried varying all kinds of parameters like number of hidden activations, dropout rate, number of layers, batch size, etc. to no avail. I initially suspected that my model might be overfitting so I lowered the hidden activations per layer drastically but saw no gain. As the epochs pass, probability for most of the predicted labels for some sample questions falls below 0.1 (Note that I'm using a callback  to predict labels for some sample questions after each epoch),

I know my post is too long but I'm at my wit's end here. You could look into my [colab notebook](https://colab.research.google.com/drive/1EOklUw7efOv69HvWKpuKVy1LSzcvTTCk) for a clearer picture.

Is my issue due to my model or my dataset? What workarounds can I adopt? I've experimented with some datasets I've compiled using the stack exchange data explorer and face more or less the same issue as above.
